testutils/mongodb.go

```diff
@@ -143,3 +147,14 @@ func BackingMongoDBClient(tb testing.TB) *mongo.Client {
 	}
 	return client
 }
+
+// BackingMongoDBClientWithOptions returns a backing MongoDB client to use with the provided options.
+func BackingMongoDBClientWithOptions(tb testing.TB, baseOptions *options.ClientOptions) *mongo.Client {
+	tb.Helper()
+	client, err := backingMongoDBClientWithOptions(baseOptions)
+	if err != nil {
+		skipWithError(tb, err)
```

Comment:
for my own understanding: why do we do this?

Reply:
i think this is so that if this fails for one specific test, itll "skip" the test with the error mesasge included and continue running other tests